src/Player.cpp

```diff
@@ -51,27 +61,35 @@ void Player::SaveGame(){
 void Player::SetPlayerData(){
     // Primarily initializes default values at the beginning of the game.
 
-	ifstream ReadData;
-	ReadData.clear();
-	ReadData.open("data.txt");
-	if (ReadData.is_open())	{
-		ReadData >> player_type;
-    ReadData.ignore();       // Ignore rest of line ready for getline
-    getline(ReadData, name);
-    ReadData >> gender;
-    ReadData >> level;
-    ReadData >> experience;
-		ReadData >> health;
-		ReadData >> arrows;
-		ReadData >> bombs;
-		ReadData >> potions;
-		ReadData >> whetstones;
-		ReadData >> weaponstrength;
-		ReadData >> coins;
+	std::ifstream ReadData("data.json");
+	if (ReadData && ReadData.peek() != EOF)
+	{
+		json SaveDataJson = json::parse(ReadData);
+		if (SaveDataJson.size() != 12)
+		{
+			cout << "savegame data length mismatch (data.json)." << endl;
+			return;
+		}
+		player_type = SaveDataJson.at("player_type").get<int>()>0? SaveDataJson.at("player_type").get<int>():1;
+		name = SaveDataJson.at("name").get<std::string>();
+		gender = SaveDataJson.at("gender").get<char>();
+		level = SaveDataJson.at("level").get<int>()>0? SaveDataJson.at("level").get<int>():1;
+		health = SaveDataJson.at("health").get<int>()>=0? SaveDataJson.at("health").get<int>():0;
+		experience = SaveDataJson.at("experience").get<int>()>=0? SaveDataJson.at("experience").get<int>():0;
+		arrows = SaveDataJson.at("arrows").get<int>()>=0? SaveDataJson.at("arrows").get<int>():0;
+		bombs = SaveDataJson.at("bombs").get<int>()>=0? SaveDataJson.at("bombs").get<int>():0;
+		potions = SaveDataJson.at("potions").get<int>()>=0? SaveDataJson.at("potions").get<int>():0;
+		whetstones = SaveDataJson.at("whetstones").get<int>()>=0? SaveDataJson.at("whetstones").get<int>():0;
+		weaponstrength = SaveDataJson.at("weaponstrength").get<int>()>=0? SaveDataJson.at("whetstones").get<int>():0;
+		coins = SaveDataJson.at("coins").get<int>()>=0? SaveDataJson.at("coins").get<int>():0;
```

Comment:
Is there a cleaner way to do this? There's lots of copy/paste programming here, maybe we can store player attributes as a list somewhere and iterate instead?